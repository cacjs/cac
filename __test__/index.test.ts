import path from 'path'
import execa from 'execa'

function fixture(file) {
  return path.relative(process.cwd(), path.join(__dirname, 'fixtures', file))
}

test('help message when no command', async () => {
  const { stdout, cmd } = await execa('node', [
    fixture('help-message/no-command.js'),
    '--help'
  ])
  expect(stdout).toMatchSnapshot(cmd)
})

test('help message for sub command', async () => {
  const { stdout, cmd } = await execa('node', [
    fixture('help-message/sub-command.js'),
    'sub',
    '--help'
  ])
  expect(stdout).toMatchSnapshot(cmd)
})

test('help message for wildcard command', async () => {
  const { stdout, cmd } = await execa('node', [
    fixture('help-message/wildcard-command.js'),
    '--help'
  ])
  expect(stdout).toMatchSnapshot(cmd)
})
