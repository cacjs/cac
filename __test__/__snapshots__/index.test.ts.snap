// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help message for sub command: node __test__/fixtures/help-message/sub-command.js sub --help 1`] = `
"
  sub-command.js 0.0.0

  USAGE

    sub-command.js sub [options]

  GLOBAL OPTIONS

    -v, --version  Display version                     [Type: boolean]
    -h, --help     Display help (You're already here)  [Type: boolean]
"
`;

exports[`help message for wildcard command: node __test__/fixtures/help-message/wildcard-command.js --help 1`] = `
"
  wildcard-command.js 0.0.0

  USAGE

    wildcard-command.js <command> [options]

  COMMANDS

    *  wildcard command

  GLOBAL OPTIONS

    -v, --version  Display version                     [Type: boolean]
    -h, --help     Display help (You're already here)  [Type: boolean]
"
`;

exports[`help message when no command: node __test__/fixtures/help-message/no-command.js --help 1`] = `
"
  no-command.js 0.0.0

  USAGE

    no-command.js [options]

  GLOBAL OPTIONS

    -v, --version  Display version                     [Type: boolean]
    -h, --help     Display help (You're already here)  [Type: boolean]
"
`;
